packages/pyright-internal/src/languageServerBase.ts

```diff
@@ -1786,33 +1786,25 @@ export abstract class LanguageServerBase implements LanguageServerInterface, Dis
             const code = this.getDiagCode(diag, rule);
             const vsDiag = Diagnostic.create(diag.range, diag.message, severity, code, this.serverOptions.productName);
 
-            if (diag.category === DiagnosticCategory.Hint) {
-                if (diag.baselined) {
-                    vsDiag.message = `Baselined: ${vsDiag.message}`;
-                }
-                if (deprecatedDiagnosticRules().includes(rule)) {
-                    // If the client doesn't support "deprecated" tags, don't report deprecated diagnostics unless it's a baselined error.
-                    if (!this.client.supportsDeprecatedDiagnosticTag && !diag.baselined) {
-                        return;
-                    }
-                    vsDiag.tags = [DiagnosticTag.Deprecated];
-                } else if ([...unreachableDiagnosticRules(), ...unusedDiagnosticRules()].includes(rule)) {
-                    // If the client doesn't support "unnecessary" tags, don't report unused code unless it's a baselined error.
-                    if (!this.client.supportsUnnecessaryDiagnosticTag && !diag.baselined) {
-                        return;
-                    }
-                    vsDiag.tags = [DiagnosticTag.Unnecessary];
+            if (diag.category === DiagnosticCategory.TaskItem) {
+                // TaskItem is not supported.
+                return;
+            }
+            if (diag.baselined) {
+                vsDiag.message = `Baselined: ${vsDiag.message}`;
+            }
+            if (deprecatedDiagnosticRules().includes(rule)) {
+                // If the client doesn't support "deprecated" tags, don't report deprecated diagnostics unless it's a baselined error.
+                if (!this.client.supportsDeprecatedDiagnosticTag && !diag.baselined) {
+                    return;
                 }
-                vsDiag.severity = DiagnosticSeverity.Hint;
-            } else {
-                assert(
-                    !diag.baselined,
-                    `a baselined diagnostic somehow had the wrong diagnostic category: ${diag.message} (${diag.category})`
-                );
```

Comment:
not sure about this part

Reply:
this check should probably still be here. i dont think its possible for this to happen but its better to be safe and check for it just in case

Reply:
I don't really understand why this is an error in the first place tbh. it's possible for `reportDeprecated` to have a different category than `DiagnosticCategory.Hint` which is fine. Can those (e.g. warning) not be baselined? I haven't used the baseline feature yet, so I am not familiar how it works

Reply:
when a diagnostic is baselined, it's always reported as a hint:

![image](https://github.com/user-attachments/assets/b62504a3-d1a1-440d-b3d8-0149204034a6)

this is so the user can still see which diagnostics have been baselined without them being reported as an error, which would defeat the purpose of it. if a baselined diagnostic were to not have the hint category, it would mean there's a bug in the baseline matching code.